### PR TITLE
General bug fixes

### PR DESCRIPTION
**Pull request by WawrzenskiPatryk**

### CODE REVIEW

File: components/Base/Button.vue [hunk 1]
```diff
@@ -48,7 +48,7 @@ const buttonLookVariants = ['fill', 'outline'];
 
 const componentType = computed(() => {
   if (props.href) return 'a';
-  if (props.to) return 'nuxt-link';
```
Comment (WawrzenskiPatryk) [inline components/Base/Button.vue:51]: For some reason this didn't generate `a` tag but `nuxt-link` in our html result. `router-link` works fine though.

File: composables/api/useProductList.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { useQuery } from 'vue-query';
 import type { Database } from '~/types/generated-types';
 import type { FullProduct } from '~/types/products';
 
-export const useProductList = (count = Infinity) => {
+export const useProductList = (count = ref(Infinity)) => {
```
Comment (WawrzenskiPatryk) [inline composables/api/useProductList.ts:5]: Passing a ref here allowed to refetch the data when we click "see more"

File: composables/api/useProductList.ts [hunk 3]
```diff
@@ -2,28 +2,33 @@ import { useQuery } from 'vue-query';
 import type { Database } from '~/types/generated-types';
 import type { FullProduct } from '~/types/products';
 
-export const useProductList = (count = Infinity) => {
+export const useProductList = (amount = ref(Infinity)) => {
```
Comment (WawrzenskiPatryk) [inline composables/api/useProductList.ts:5]: Passing a `ref` here allowed to refetch the data when we click "see more" on the `/fleet` page

File: composables/api/useProductList.ts [hunk 4]
```diff
@@ -2,28 +2,33 @@ import { useQuery } from 'vue-query';
...
-        .limit(count);
+        .select('*, product_specifications (*)', { count: 'exact' })
+        .limit(amount.value);
 
       if (error) {
         console.error(error);
         throw new Error(error.message);
       }
+
+      maxAmount.value = count;
```
Comment (WawrzenskiPatryk) [inline composables/api/useProductList.ts:22]: I used `count` returned from database to conditionally show "see more" button on the `/fleet` page.

**Review by elmir-ismayilzada — APPROVED** [review]
Looks good!

**Pull request merged**